### PR TITLE
Waiting for data adapter/cache creation in bundle importer.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

Before this change, lookup data adapters, caches and tables were created
sequentially during content pack import, leading to errors when lookup
tables were created before caches or data adapters were successfully
created and started. Imported lookup tables were usable only after the
server was restarted at least once after a content pack import.

After this change, the bundle importer waits for the successful creation
of data adapters/caches before creating lookup tables.

Fixes Graylog2/graylog-plugin-threatintel#57.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/bundles/BundleImporter.java [hunk 1]
```diff
@@ -709,9 +729,19 @@ private void createLookupCaches(String bundleId, Set<LookupCacheBundle> lookupCa
...
+
+        if (!createdLookupCaches.isEmpty()) {
+            clusterBus.post(CachesUpdated.create(createdLookupCaches.keySet()));
+        }
+
+        final Collection<LookupCache> caches = lookupTableService.getCaches(createdLookupCaches.keySet());
+        final CountDownLatch latch = new CountDownLatch(caches.size());
+        caches.forEach(c -> c.addListener(new LatchUpdaterListener(latch), scheduler));
+
+        latch.await(30, TimeUnit.SECONDS);
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/bundles/BundleImporter.java]: We should at least log a short message on either INFO or WARN if the timeout expired (which means that not all caches were refreshed).

File: graylog2-server/src/main/java/org/graylog2/bundles/BundleImporter.java [hunk 2]
```diff
@@ -722,5 +752,15 @@ private void createLookupDataAdapters(String bundleId, Set<LookupDataAdapterBund
...
+
+        if (!createdLookupDataAdapters.isEmpty()) {
+            clusterBus.post(DataAdaptersUpdated.create(createdLookupDataAdapters.keySet()));
+        }
+
+        final Collection<LookupDataAdapter> dataAdapters = lookupTableService.getDataAdapters(createdLookupDataAdapters.keySet());
+        final CountDownLatch latch = new CountDownLatch(dataAdapters.size());
+        dataAdapters.forEach(da -> da.addListener(new LatchUpdaterListener(latch), scheduler));
+
+        latch.await(30, TimeUnit.SECONDS);
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/bundles/BundleImporter.java]: We should at least log a short message on either INFO or WARN if the timeout expired (which means that not all caches were refreshed).

Comment (dennisoelkers) [reply graylog2-server/src/main/java/org/graylog2/bundles/BundleImporter.java]: ✅

File: graylog2-server/src/main/java/org/graylog2/bundles/BundleImporter.java [hunk 3]
```diff
@@ -738,10 +737,14 @@ private void createLookupCaches(String bundleId, Set<LookupCacheBundle> lookupCa
         final CountDownLatch latch = new CountDownLatch(caches.size());
         caches.forEach(c -> c.addListener(new LatchUpdaterListener(latch), scheduler));
 
-        latch.await(30, TimeUnit.SECONDS);
+        try {
+            latch.await(30, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            LOG.warn("Starting imported Lookup Table Caches did not finish within 30 seconds. A server restart might be required for imported Lookup Tables to function.");
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/bundles/BundleImporter.java]: The `InterruptedException` is not thrown when the latch's timeout expires but only when the thread is being interrupted.

If the timeout expired, [`CountDownLatch#await()`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CountDownLatch.html#await-long-java.util.concurrent.TimeUnit-) will return `false`.

File: graylog2-server/src/main/java/org/graylog2/bundles/BundleImporter.java [hunk 4]
```diff
@@ -761,6 +764,10 @@ private void createLookupDataAdapters(String bundleId, Set<LookupDataAdapterBund
         final CountDownLatch latch = new CountDownLatch(dataAdapters.size());
         dataAdapters.forEach(da -> da.addListener(new LatchUpdaterListener(latch), scheduler));
 
-        latch.await(30, TimeUnit.SECONDS);
+        try {
+            latch.await(30, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            LOG.warn("Starting imported Lookup Table Data Adapters did not finish within 30 seconds. A server restart might be required for imported Lookup Tables to function.");
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/bundles/BundleImporter.java]: The `InterruptedException` is not thrown when the latch's timeout expires but only when the thread is being interrupted.

If the timeout expired, [`CountDownLatch#await()`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CountDownLatch.html#await-long-java.util.concurrent.TimeUnit-) will return `false`.

File: graylog2-server/src/main/java/org/graylog2/bundles/BundleImporter.java (hunk 3 above)
Comment (dennisoelkers) [reply graylog2-server/src/main/java/org/graylog2/bundles/BundleImporter.java]: ✅ (Should read javadocs properly)

File: graylog2-server/src/main/java/org/graylog2/bundles/BundleImporter.java (hunk 4 above)
Comment (dennisoelkers) [reply graylog2-server/src/main/java/org/graylog2/bundles/BundleImporter.java]: ✅ (Should read javadocs properly)

**Pull request merged**